### DISCUSSION TITLE
How to add call options in grpc client?

### DISCUSSION THREAD

**Discussion by alienzhangyw** [thesayyn/protoc-gen-ts#73] [Q&A] [answered]
There [CallOptions](https://grpc.github.io/grpc/node/grpc.Client.html#~CallOptions__anchor) in grpc calls. 
How can I use them in grpc client calls?
Current example:
https://github.com/thesayyn/protoc-gen-ts/blob/509e11c28f0c8f242448c697670df4370d82050d/examples/grpc/client.ts#L22

Should I write like this?
``` ts
const options = { deadline: Date.now() + 5000 };
client["add"](payload, options, callback);
```
[reference: How to set timeout of grpc client in node.js
](https://stackoverflow.com/a/46590945)

**Comment by thesayyn** [comment, marked as answer]
You can pass it as a second argument. protoc-gen-ts does not do anything special thing for the grpc client

**Comment by thesayyn** [comment]
there are some upcoming breaking changes for the server implementation though.

https://github.com/thesayyn/protoc-gen-ts/blob/32cf5f5bcd92aecb356532ee5bf8042202f68f95/docs/rpc.md

**Reply by alienzhangyw** [reply]
The client instance doesn't have type intelligence for request methods right now.

**Reply by thesayyn** [reply]
that's true. i am trying to find time to work on it but couldn’t find any yet

**Reply by thesayyn** [reply]
i am making progress on typed service clients on #72
